lib/puppet/provider/package/yum.rb

```diff
@@ -155,7 +155,18 @@ def install
     self.debug "Ensuring => #{should}"
     operation = :install
 
+    current_package = self.query
```

Comment:
I'm not a fan of doing this when it's not necessary (i.e. `when true, false, Symbol`).